ibtk/src/lagrangian/LIndexSetData.cpp

```diff
@@ -62,8 +62,8 @@ namespace IBTK
 /////////////////////////////// PUBLIC ///////////////////////////////////////
 
 template <class T>
-LIndexSetData<T>::LIndexSetData(const Box<NDIM>& box, const IntVector<NDIM>& ghosts)
-    : LSetData<T>(box, ghosts),
+LIndexSetData<T>::LIndexSetData(Box<NDIM> box, IntVector<NDIM> ghosts)
+    : LSetData<T>(std::move(box), std::move(ghosts)),
```

Comment:
@boyceg I made these kind of changes without thinking about it too much, but now I'm not sure. This requires `Box` and `IntVector` to have correct copy constructors, right? The compiler should generate them automatically, but can we assume they work correctly?

Reply:
`Box<DIM>` and `IntVector<DIM>` both have proper copy constructors.

Reply:
I mean, they should have proper move constructors.

Reply:
They do not have explicit move constructors.

I do not believe that they will get implicit move constructors --- c.f. https://stackoverflow.com/questions/13344800/can-a-move-constructor-be-implicit.

So this seems dangerous.

Reply:
...and this is one potential issue with defaulting to use pass-by-value + `std::move` where possible --- you have to know what has a move constructor and what doesn't.

I wonder if it would be better to stick with preferring to use pass-by-const-reference for now. (Paging @drwells for an opinion. ;-)

Reply:
Let me take a look.

Reply:
Since `Box<NDIM>` (in SAMRAI 2 and SAMRAI 3) has a non-implicit copy constructor the compiler will not generate an implicit move constructor. Hence, this will ultimately just call the copy constructor anyway (moving duplicates the data while leaving the original object in an unspecified, but valid, state; copying does exactly that).

I don't think this matters a lot for 99% of SAMRAI types since most of them just contain a couple integers. A `Box` contains two `Index`es, a `BlockId` (an `unsigned integer`), a `BoxId` (a couple integers, signd and unsigned), a `bool`, and an `EmptyBoxState` (an `enum`): there is no advantage to writing custom move constructors for that anyway.